nixos/modules/services/web-servers/garage.nix

```diff
@@ -75,7 +75,25 @@ in
       source = configFile;
     };
 
-    environment.systemPackages = [ cfg.package ]; # For administration
+    # For administration
+    environment.systemPackages = if cfg.environmentFile != null
+      then [(
+        # reads the env vars from `cfg.environmentFile` to the garage environment
+        # makes working with the cli quite a bit nicer
+        pkgs.writeScriptBin "garage" ''
+          # make it so all future variables set are automatically exported as environment variables
+          set -a
+
+          # source the set environmentFile (since systemd EnvironmentFile is supposed to be a minor subset of posix sh parsing) (with shell arg escaping to avoid quoting issues)
+          . ${lib.escapeShellArg cfg.environmentFile}
+
+          # exec the program with quoted args (also with shell arg escaping for the program path to avoid quoting issues there)
```

Comment:
Comment says there's shell escaping for the program path, but the implementation doesn't do that.

Reply:
woops yeah it's fixed now.